sql/2021/security/mixed_content.sql

```diff
@@ -0,0 +1,22 @@
+#standardSQL
+# Prevalence of landing pages over HTTPS that initiate at least one request over HTTP and distribution over ranking.
+SELECT
+  COUNT(DISTINCT page) AS total_pages,
+  COUNT(DISTINCT(CASE WHEN REGEXP_CONTAINS(page, r'https://.*') AND REGEXP_CONTAINS(url, r'http://.*') THEN page END)) AS count_http_requests,
+  COUNT(DISTINCT(CASE WHEN REGEXP_CONTAINS(page, r'https://.*') AND REGEXP_CONTAINS(url, r'http://.*') THEN page END)) / COUNT(DISTINCT page) AS pct_http_requests,
+  COUNT(DISTINCT(CASE WHEN REGEXP_CONTAINS(page, r'https://.*') AND REGEXP_CONTAINS(url, r'http://.*') AND rank = 1000 THEN page END)) AS count_top_1k,
+  COUNT(DISTINCT(CASE WHEN REGEXP_CONTAINS(page, r'https://.*') AND REGEXP_CONTAINS(url, r'http://.*') AND rank = 1000 THEN page END)) / COUNT(DISTINCT page) AS pct_top_1k,
+  COUNT(DISTINCT(CASE WHEN REGEXP_CONTAINS(page, r'https://.*') AND REGEXP_CONTAINS(url, r'http://.*') AND rank = 10000 THEN page END)) AS count_top_10k,
+  COUNT(DISTINCT(CASE WHEN REGEXP_CONTAINS(page, r'https://.*') AND REGEXP_CONTAINS(url, r'http://.*') AND rank = 10000 THEN page END)) / COUNT(DISTINCT page) AS pct_top_10k,
+  COUNT(DISTINCT(CASE WHEN REGEXP_CONTAINS(page, r'https://.*') AND REGEXP_CONTAINS(url, r'http://.*') AND rank = 100000 THEN page END)) AS count_top_100k,
+  COUNT(DISTINCT(CASE WHEN REGEXP_CONTAINS(page, r'https://.*') AND REGEXP_CONTAINS(url, r'http://.*') AND rank = 100000 THEN page END)) / COUNT(DISTINCT page) AS pct_top_100k,
+  COUNT(DISTINCT(CASE WHEN REGEXP_CONTAINS(page, r'https://.*') AND REGEXP_CONTAINS(url, r'http://.*') AND rank = 1000000 THEN page END)) AS count_top_1m,
+  COUNT(DISTINCT(CASE WHEN REGEXP_CONTAINS(page, r'https://.*') AND REGEXP_CONTAINS(url, r'http://.*') AND rank = 1000000 THEN page END)) / COUNT(DISTINCT page) AS pct_top_1m,
+  COUNT(DISTINCT(CASE WHEN REGEXP_CONTAINS(page, r'https://.*') AND REGEXP_CONTAINS(url, r'http://.*') AND rank = 10000000 THEN page END)) AS count_top_10m,
```

Comment:
These ranks need to be `<=` rather than `=` as rank 10,000 is actually only the next 9,000 pages and not inclusive of the top 1,000 pages.

Also is this the best way of presenting this? Normally would do an `UNNEST` - see https://github.com/HTTPArchive/almanac.httparchive.org/blob/main/sql/2021/pwa/sw_adoption_over_time_ranking.sql as an example.